test/functional/legacy/listlbr_utf8_spec.lua

```diff
@@ -1,57 +1,116 @@
 -- Test for linebreak and list option in utf-8 mode
 
 local helpers = require('test.functional.helpers')
+local clear = helpers.clear
+local dedent = helpers.dedent
+local eq = helpers.eq
+local expect = helpers.expect
+local feed = helpers.feed
+local nvim = helpers.meths
 local source = helpers.source
-local clear, expect = helpers.clear, helpers.expect
+
+local function expect_line(line)
+  local expected = dedent(line)
+  local actual = nvim.get_var('line')
+  eq(expected, actual)
+end
 
 describe('linebreak', function()
-  setup(clear)
 
-  it('is working', function()
+  before_each(function()
+    clear()
+    
     source([[
       set wildchar=^E
       10new
       vsp
       vert resize 20
-      put =\"\tabcdef hijklmn\tpqrstuvwxyz\u00a01060ABCDEFGHIJKLMNOP \"
-      norm! zt
       set ts=4 sw=4 sts=4 linebreak sbr=+ wrap
+      
       fu! ScreenChar(width, lines)
-      	let c=''
-      	for j in range(1,a:lines)
-      	    for i in range(1,a:width)
-      	    	let c.=nr2char(screenchar(j, i))
-      	    endfor
-                 let c.="\n"
-      	endfor
-      	return c
+        let c=''
+        for j in range(1,a:lines)
+            for i in range(1,a:width)
+              let c.=nr2char(screenchar(j, i))
+            endfor
+            let c.="\n"
+        endfor
+        return c
       endfu
+      
       fu! DoRecordScreen()
-      	wincmd l
-      	$put =printf(\"\n%s\", g:test)
-      	$put =g:line
-      	wincmd p
+        wincmd l
+        $put =printf(\"\n%s\", g:test)
+        $put =g:line
+        wincmd p
       endfu
-      let g:test ="Test 1: set linebreak + set list + fancy listchars"
-      exe "set linebreak list listchars=nbsp:\u2423,tab:\u2595\u2014,trail:\u02d1,eol:\ub6"
+      ]])
+  end)
+
+  it('Test 1: set linebreak + set list + fancy listchars', function()
+    source([[
+      put =\"\tabcdef hijklmn\tpqrstuvwxyz\u00a01060ABCDEFGHIJKLMNOP \"
+      norm! zt
+      set linebreak list listchars=nbsp:␣,tab:▕—,trail:ˑ,eol:¶
       redraw!
-      let line=ScreenChar(winwidth(0),4)
-      call DoRecordScreen()
-      let g:test ="Test 2: set nolinebreak list"
-      set list nolinebreak
+      let line = ScreenChar(winwidth(0),4)
```

Comment:
This `let line = ...` idiom doesn't seem much different than using `call append()` with the existing `helpers.expect()`. What's the benefit of this change?

Also it is generally recommended not to drastically change the legacy tests, because it makes merging harder. Though breaking it up into `it()` is usually helpful.